Sync/SyncStateMachine.swift

```diff
@@ -276,7 +291,8 @@ open class BaseSyncState: SyncState {
         })
     }
 
-    public init(client: Sync15StorageClient, scratchpad: Scratchpad, token: TokenServerToken) {
+    public init(prefs: Prefs, client: Sync15StorageClient, scratchpad: Scratchpad, token: TokenServerToken) {
```

Comment:
I think passing `Prefs` in here is fundamentally wrong. I think you should consider modeling this flow (and persisting these choices) via the `Scratchpad`: that is, when entering the state machine, we might or might not have some pending engine enablements or disablements that need to be reflected.

There are a few places that applies:

- If we encounter a blank server, we apply those pending changes to the previous persisted state, and upload.
- If we encounter a server with a conflicting meta/global, we resolve the conflict.
- If we encounter a server with an unchanged meta/global, we produce a new m/g and upload it, re-entering the flow in the expected place.